sdk/storage/azfile/directory/client.go

```diff
@@ -45,9 +50,12 @@ func NewClientWithSharedKeyCredential(directoryURL string, cred *SharedKeyCreden
 	authPolicy := exported.NewSharedKeyCredPolicy(cred)
 	conOptions := shared.GetClientOptions(options)
 	conOptions.PerRetryPolicies = append(conOptions.PerRetryPolicies, authPolicy)
```

Comment:
This really should be.
```go
	plOpts := runtime.PipelineOptions{
		PerRetry: []policy.Policy{authPolicy},
	}
	azClient, err := azcore.NewClient(shared.DirectoryClient, exported.ModuleVersion, plOpts, &conOptions.ClientOptions)
```
It was missed in previous reviews.  I first noticed this in `azblob` earlier this week.  If you don't want to fix it now, can you please open a tracking issue?

Reply:
Added the auth policy in the `PerRetry` field of `runtime.PipelineOptions`.

Reply:
@jhendrixMSFT, I have resolved the other comments. Can you please review and let me know if there are more changes needed.